src/securejoin.rs

```diff
@@ -540,7 +573,7 @@ pub(crate) async fn handle_securejoin_handshake(
                 let bob = context.bob.read().await;
                 let scan = bob.qr_scan.as_ref();
                 scan.is_none()
-                    || bob.expects != DC_VC_AUTH_REQUIRED
+                    || !matches!(bob.expects, SecureJoinStep::AuthRequired)
```

Comment:
How about deriving `PartialEq` for enum instead? `!matches!` is harder to read than `!=`.